packages/ui/src/ui/pages/navigation/NavigationDescription/NavigationDescriptionOverview.tsx

```diff
@@ -0,0 +1,85 @@
+import React, {Dispatch, FC, SetStateAction, useCallback, useRef} from 'react';
+import {useDispatch, useSelector} from 'react-redux';
+import {Flex} from '@gravity-ui/uikit';
+
+import {getPath} from '../../../store/selectors/navigation';
+import {yt as ytApi} from '../../../store/api/yt';
+
+import {EditButtons} from './EditButtons';
+import {SwitchDescription} from './SwitchDescription';
+import {makeAnnotationFetchParams, useUpdateAnnotation} from './hooks';
+
+interface Annotation {
+    annotation?: string;
+    annotationPath?: string;
+}
+
+interface NavigationDescriptionOverviewProps {
+    isEditing: boolean;
+    setIsEditing: Dispatch<SetStateAction<boolean>>;
+    ytAnnotation: Annotation;
+    externalAnnotation: Annotation;
+    annotationSwitch: {
+        showSwitch?: boolean;
+        showExternalDescription: boolean;
+        setShowExternalDescription: Dispatch<SetStateAction<boolean>>;
+    };
+}
+
+export const NavigationDescriptionOverview: FC<NavigationDescriptionOverviewProps> = ({
+    ytAnnotation: {annotation},
+    externalAnnotation: {annotationPath},
+    isEditing,
+    setIsEditing,
+    annotationSwitch: {showSwitch, showExternalDescription, setShowExternalDescription},
+}) => {
+    const dispatch = useDispatch();
+
+    const path = useSelector(getPath);
+
+    const oldValue = useRef<string>(annotation || '');
+
+    const [updateFn, {isLoading}] = useUpdateAnnotation();
+
+    const handleEditClick = useCallback(() => {
+        oldValue.current = annotation || '';
+        setIsEditing(true);
+    }, [annotation, setIsEditing]);
+
+    const handleCancelClick = useCallback(() => {
+        dispatch(
+            ytApi.util.updateQueryData('fetchBatch', makeAnnotationFetchParams(path), () => [
```

Comment:
We don't need to restore data if we keep user input separately from loaded data.